PWGHF/HFC/Tasks/taskCorrelationDMesonPairs.cxx

```diff
@@ -0,0 +1,569 @@
+      registry.fill(HIST("hCorrel2DVsPtSidebandsMCRecBkgBkg"), deltaPhi, deltaEta, ptCand1, ptCand2);
+    }
+  }
+
+  template <typename T>
+  void analyseData(const T& pairEntries)
+  {
+    for (auto& pairEntry : pairEntries) {
+      if (pairEntry.isData() == 1) {
+        // define variables for widely used quantities
+        double deltaPhi = pairEntry.deltaPhi();
+        double deltaEta = pairEntry.deltaEta();
+        double ptCand1 = pairEntry.ptCand1();
+        double ptCand2 = pairEntry.ptCand2();
+        double massCand1 = pairEntry.mCand1();
+        double massCand2 = pairEntry.mCand2();
+        double yCand1 = pairEntry.yCand1();
+        double yCand2 = pairEntry.yCand2();
+
+        int pTBinCand1 = o2::analysis::findBin(binsPtCorrelations, ptCand1);
+        int pTBinCand2 = o2::analysis::findBin(binsPtCorrelations, ptCand2);
+
+        // fill 2D invariant mass plots
+        registry.fill(HIST("hMass2DCorrelationPairs"), massCand1, massCand2, ptCand1, ptCand2);
+
+        // reject entries outside pT ranges of interest
+        if (pTBinCand1 == -1 || pTBinCand2 == -1) { // at least one particle outside accepted pT range
+          continue;
+        }
+
+        // check if correlation entry belongs to signal region, sidebands or is outside both, and fill correlation plots
+        if (massCand1 > signalRegionInner->at(pTBinCand1) && massCand1 < signalRegionOuter->at(pTBinCand1) && massCand2 > signalRegionInner->at(pTBinCand2) && massCand2 < signalRegionOuter->at(pTBinCand2)) {
+          // in signal region
+          registry.fill(HIST("hCorrel2DVsPtSignalRegion"), deltaPhi, deltaEta, ptCand1, ptCand2);
+          registry.fill(HIST("hCorrel2DPtIntSignalRegion"), deltaPhi, deltaEta);
+          registry.fill(HIST("hDeltaEtaPtIntSignalRegion"), deltaEta);
+          registry.fill(HIST("hDeltaPhiPtIntSignalRegion"), deltaPhi);
+          registry.fill(HIST("hDeltaYPtIntSignalRegion"), yCand2 - yCand1);
+          registry.fill(HIST("hDeltaPtDDbarSignalRegion"), ptCand2 - ptCand1);
+          registry.fill(HIST("hDeltaPtMaxMinSignalRegion"), std::abs(ptCand2 - ptCand1));
+        }
+
+        if ((massCand1 > sidebandLeftInner->at(pTBinCand1) && massCand1 < sidebandLeftOuter->at(pTBinCand1) && massCand2 > sidebandLeftInner->at(pTBinCand2) && massCand2 < sidebandRightOuter->at(pTBinCand2)) ||
+            (massCand1 > sidebandRightInner->at(pTBinCand1) && massCand1 < sidebandRightOuter->at(pTBinCand1) && massCand2 > sidebandLeftInner->at(pTBinCand2) && massCand2 < sidebandRightOuter->at(pTBinCand2)) ||
+            (massCand1 > sidebandLeftInner->at(pTBinCand1) && massCand1 < sidebandRightOuter->at(pTBinCand1) && massCand2 > sidebandLeftInner->at(pTBinCand2) && massCand2 < sidebandLeftOuter->at(pTBinCand2)) ||
+            (massCand1 > sidebandLeftInner->at(pTBinCand1) && massCand1 < sidebandRightOuter->at(pTBinCand1) && massCand2 > sidebandRightInner->at(pTBinCand2) && massCand2 < sidebandRightOuter->at(pTBinCand2))) {
```

Comment:
Avoid repetition and simplify the condition.
Btw, why would you accept mass between `sidebandLeftInner` and `sidebandRightOuter`?